src/flow-spec/IResource.ts

```diff
@@ -7,7 +7,7 @@ export interface IResourceValue {
   modes: SupportedMode[]
   /**
    * Value will accept alphanumerics, white spaces, file path, expressions, conditions,
```

Comment:
Is there anything this could hold that would be considered _invalid_?  (If not, do we need a regex validation at all?  Can this just be unconstrained, as any Unicode string?)

Reply:
\p{L} matches a single code point in the category “letter”

So this means things like unicode emoji would not be supported. And any special characters not whitelisted here would also not be included.

Reply:
From a Flow Spec perspective, things like Unicode emoji should definitely be allowed. 

Are you guys aware of any characters that should _not_ be allowed in message content (e.g. because they would break something in our parsing or runtime?)

Can this just be `.*`? (Or no validation constraints at all, if that is an option?)

Reply:
I believe the only special characters that are omitted currently are tilde, ampersand, brackets, and pipe. 

Barring those (and some unicode emoji) this is pretty permissive in its current state. At this point, your consideration that we might even omit regex at all might be valid.

Reply:
Good observation @markboots!
I think we have missed this since the beginning
> From a Flow Spec perspective, things like Unicode emoji should definitely be allowed.

In fact, this is the history of resource.value restriction:
History of resource.value restriction:
- make resource.value required (#4de9f3da) > `@pattern ^[a-zA-Z_]\w*$`
- Value will accept text and file path (eg: my-path/the_file.mp3) (#81e7c494) > `@pattern ^[\w\\\/.-]+$`
- Value will accept alphanumerics, white spaces, file path, expressions, conditions, (#415899b8) > `@pattern ^[\w\s\\\/@:,.+*^<>=()"'-]+$`

As you see, at the beginning we just thought it would be a **required** text. But, following Mark's input here, I think we should just abandon the regex and just check it should have at least a value with `@minLength 1`

Your thoughts @seifertk, @markboots, @GrimFunction ?
cc: @object-required, @bulatgab

Reply:
Thanks @safydy-r , I agree on removing the regex constraint completely. Flow resources can be any unicode string, so there's no need for a regex check. Thanks for the history! Probably how we got here is the initial attempt at making a resource value required (#4de9f3da) assumed a regex approach and a different understanding of what the content could be.

Reply:
fyi: I created a new PR for this to address the decision in current thread https://github.com/FLOIP/flow-runner/pull/100